canonical/maas-ui
# feat: Global side navigation MAASENG-1272

## Done

- Created new side navigation component
- Wrote integration tests
- Removed old header-style navigation

## Todo

- [ ] Make nav collapsible
- [x] Fix weird issue with growing and shrinking on hover

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Open MAAS-UI
- Ensure the new side navigation renders
- Ensure all of the links work as intended
- Ensure link-page highlighting works as intended
- Click your username
- Click "Log out"
- Ensure you are logged out

## Fixes

Fixes [MAASENG-1272](https://warthogs.atlassian.net/jira/software/c/projects/MAASENG/boards/302?modal=detail&selectedIssue=MAASENG-1272)

## Screenshots

### Before
![image](https://user-images.githubusercontent.com/35104482/215771724-a0f8a1c5-2c4d-40a3-9879-82c7349e1545.png)

### After
![image](https://user-images.githubusercontent.com/35104482/215771798-adc9b51b-c3aa-49ad-b321-956ffdc9dd63.png)

#### With theme
![image](https://user-images.githubusercontent.com/35104482/215772980-07daffe8-0da8-4491-a552-1a8a5213aaf4.png)

#### Contextual menu for user
![image](https://user-images.githubusercontent.com/35104482/215826725-992123da-dd5e-4333-a099-7fef2c3f295b.png)

[MAASENG-1272]: https://warthogs.atlassian.net/browse/MAASENG-1272?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
